frontend/public/components/utils/request-size-input.tsx

```diff
@@ -28,8 +28,10 @@ export const RequestSizeInput: React.FC<RequestSizeInputProps> = ({
   };
 
   const changeValueBy = (changeBy: number) => {
-    setValue(value + changeBy);
-    onChange({ value: value + changeBy, unit });
+    // When default defaultRequestSizeValue is not set, value becomes NaN and increment decrement buttons of NumberSpinner doesn't work.
+    const newValue = !isNaN(value) ? value + changeBy : 0 + changeBy;
```

Comment:
`Number.isFinite` is a better check here since it handles some other cases.

```suggestion
    const newValue = !Number.isFinite(value) ? value + changeBy : 0 + changeBy;
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite

Reply:
Updated.